lua/telescope/_extensions/project_actions.lua

```diff
@@ -0,0 +1,61 @@
+local builtin = require("telescope.builtin")
+local actions = require("telescope.actions")
+local transform_mod = require('telescope.actions.mt').transform_mod;
+
+local project_actions = {}
+
+local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.txt'
+
+project_actions.add_project = function()
+	local git_root = vim.fn.systemlist("git -C " .. vim.loop.cwd() .. " rev-parse --show-toplevel")[1]
+	if not git_root then
+		print('Telescope-project adds projects based on a git root. No project found. Please refer to documentation to add projects manually')
+		return
+	end
+
+	local project_title = git_root:match("[^/]+$")
+	local project_to_add = project_title .. "=" .. git_root
+
+	local file = assert(io.open(project_dirs_file, "a"), "No project file exists")
+	if not file then
+		return
+	end
```

Comment:
Will assert not stop execution anyway. Is the if return not redundant?
Also you can just create the file if not here.

Reply:
@Conni2461 Good point. I am creating the file if it's not there in a recent commit. I put this assert just in case for any reason the file failed to save. I'll delete this code